deploy/deploy.py

```diff
@@ -167,6 +174,36 @@ def _install_or_update_package(
         )
 
 
+def _grant_sundeck_db_access(cur, sundeck_db: str):
+    print("Running grants to database SUNDECK for application package.")
+    # Grant REFERENCE_USAGE on SUNDECK to the application package
+    # Grant access to the view from sundeck
+    cur.execute(
+        f"""
+    BEGIN
+        CREATE SCHEMA IF NOT EXISTS "{APPLICATION_PACKAGE}".SHARING;
+
+        show databases;
+        let db_exists boolean := (SELECT "name" is not null FROM TABLE(RESULT_SCAN(LAST_QUERY_ID())) WHERE "name" = '{sundeck_db}');
+        if (db_exists) then
+            GRANT REFERENCE_USAGE ON DATABASE {sundeck_db} TO SHARE IN APPLICATION PACKAGE "{APPLICATION_PACKAGE}";
+            -- Create a view in the application package that is filtered to CURRENT_ACCOUNT(). It is critical that the
+            -- filtering is done here to ensure a user only sees their own history.
+            CREATE OR REPLACE VIEW "{APPLICATION_PACKAGE}".SHARING.GLOBAL_QUERY_HISTORY AS SELECT * FROM {sundeck_db}.INTERNAL.GLOBAL_QUERY_HISTORY
+                WHERE UPPER(SNOWFLAKE_ACCOUNT_LOCATOR) = UPPER(CURRENT_ACCOUNT());
```

Comment:
its not clear to me how this works...we are executing a statement with `CURRENT_ACCOUNT` in our account. This will only share our data?

Reply:
> we are executing a statement with CURRENT_ACCOUNT in our account. This will only share our data?

I had expected the current_account would be late-bound to the current_account when the view is queried, but I have not actually tested that is happening.

Reply:
might be worth spinnign up an app and sharing w/ anothe rone of our accounts. I don't like that the security stuff is split across two views and I think it would be good to test, verify and consoilidate down

Reply:
Pushed down region and cloud via the context functions into the application package view. The tenant_id filter is still happening in the setup script, but I'm not sure of a way around that. An adventurous user could potentially see queries executed in their Snowflake account by a different Sundeck account. I could imagine scenarios where that is "very bad". I'll take another look and see if I can come up with a way to push the Sundeck account info down, too.